samples/nrf_desktop/src/modules/config.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdlib.h>
+
+#include <zephyr.h>
+#include <settings/settings.h>
+
+#include "event_manager.h"
+#include "config_event.h"
+
+#define MODULE config
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_LOG_LEVEL);
+
+#define KEY_LEN 9
```

Comment:
9 seems short... e.g. `power_manager/xy` will not fit in

Reply:
`config/x` is all we need right now. If other modules are added it will be increased. It will be observed immediately on log messages (you would see `inf: Store in power_ma`)

Reply:
Other thing to note is that all other modules use full names (as literals) when they save to settings, e.g. `settings_save_one(MODULE_NAME "/taddress", ...)`. Config module is the only one that builds keys from integer value because it does not own the data.

Reply:
this may be hard to notice; we should assert name fits

Reply:
done